smarts/core/utils/episodes.py

```diff
@@ -74,7 +74,7 @@ def record_step(self, observations=None, rewards=None, dones=None, infos=None):
                 observations, rewards, dones, infos
             )
 
-        if dones.get("__all__", False) and infos is not None:
+        if dones and infos is not None:
```

Comment:
I think `dones` here is a `dict`. Why was this changed?

Reply:
From the issue #1279 one of the requests was that `done` be related to the environment being done rather than for individual agents. I have changed `done` to represent `dones["__all__"]` instead to reflect the request. See https://github.com/huawei-noah/SMARTS/pull/1374/files#diff-7f6d3d7e26af4d9bdf221b3951430a399520fd204bdf830e475813052c2c1d28R333

Reply:
This is correct for `Gym` specific environments. When we get around to `RLlib` style environments that may be different.